README.md

```diff
@@ -8,39 +8,56 @@ will likely not be included in this repository, such as custom webhooks, etc.
 
+## Setup dev environment
 
-### Setup dev environment
-#### Requirements
+### Requirements
 You will need to install:
 * [docker](https://www.docker.com/) -> as the container engine for everything.
 * [docker-compose](https://docs.docker.com/compose/) -> for harbor (docker registry).
 * [minikube](https://minikube.sigs.k8s.io/docs/) -> as dev kubernetes installation.
 
-#### Run harbor
-You can install it with the helper script:
-* `utils/get_harbor.sh`
-After that you can use docker-compose to run the whole harbor system (see the output of the script for the exact command).
-
-Once harbor is running, you will need to make sure that it's setup (user project created and such) so run:
-- `utils/setup_harbor.py`
-
 ### Setup minikube
 We will need to get a specific k8s version (1.21.8 is the current toolforge version when writing this, you might want to double check):
  - `minikube start --kubernetes-version=v1.21.8`
 
+### Find your Docker IP
+You will need to reach the Docker daemon on your host from inside Minikube using
+an IP (this is a limitation of Tekton that doesn't support hostnames unless you
+have a valid SSL certificate).
+
+The IP to use will vary depending on your machine, but you can usually find it
+by running:
+- `minikube ssh`
+- `grep host.minikube.internal /etc/hosts`
+
+### Setup Harbor
+You can install Harbor with the helper script:
+- `utils/get_harbor.sh DOCKER_IP` (use the Docker IP from the previous step)
+
+After that you can use docker-compose to run the whole harbor system:
+- `docker-compose -f /Users/fran/wmf/buildservice/.harbor/harbor/docker-compose.yml up -d`
+
+If the docker-compose command exits with an error, try running it again (there
+is a race condition that sometimes makes it fail on the first run).
+
+Once harbor is running, you will need to make sure that it's setup (user project created and such) so run:
+- `utils/setup_harbor.py`
+
 ### Setup admission controller (optional)
-If you want to do a full stack test, you'll need to deploy the buildpack admission controller too, for that follow the instructions [here](https://github.com/toolforge/buildpack-admission-controller).
+If you want to do a full stack test, you'll need to deploy the buildpack
+admission controller too, for that follow the instructions
+[here](https://github.com/toolforge/buildpack-admission-controller).
+
 **NOTE**: might be faster to build the buildpack admission controller image locally instead of pulling it.
 
 ### Deploying
-You should change the 'CHANGEME' values for the ones specific to your environment, you can do a quick:
-```
-git grep CHANGEME
-```
+Create a file `deploy/devel/harbor-ip-patch.yaml`, copying the content from
+`harbor-ip-patch.yaml.example` and replacing HARBOR_IP with your Docker IP (see
+the section "Find your Docker IP" above in this document).
 
 If you want to check first what would be deployed, you can run:
-- `kubectl kustomize deploy/base-tekton | vim -`
-- `kubectl kustomize deploy/devel | vim -`
```

Comment:
nit: Oooh, no love for vim? xd

Reply:
Lots of love for VIM, but I'd leave it up to the user to choose between `vim`, `less`, or other options 😄